db/migrate/20240612063735_add_field_type_enum_to_user_fields.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class AddFieldTypeEnumToUserFields < ActiveRecord::Migration[7.0]
+  def change
+    add_column :user_fields, :field_type_enum, :integer
+
+    up_only do
+      execute(<<~SQL)
+        UPDATE user_fields
+        SET field_type_enum =
+          CASE
+            WHEN field_type = 'text' THEN 0
+            WHEN field_type = 'confirm' THEN 1
+            WHEN field_type = 'dropdown' THEN 2
+            WHEN field_type = 'multiselect' THEN 3
```

Comment:
Verified the string values by creating a field of each type locally. I have not found any plugin in `discourse` or `discourse-org` that extends this.